src/librustc/session/mod.rs

```diff
@@ -504,6 +518,33 @@ impl Session {
         println!("Total time spent decoding DefPath tables:      {}",
                  duration_to_secs_str(self.perf_stats.decode_def_path_tables_time.get()));
     }
+
+    /// We want to know if we're allowed to do an optimization for crate crate.
```

Comment:
Weirdly I'm going to go with yes, though crate foo might be a better choice for this phrasing.  It's something I would write, though, with the intent that the redundant crate is a variable.